presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java

```diff
@@ -55,35 +51,32 @@ public TableScanNode(
             @JsonProperty("id") PlanNodeId id,
             @JsonProperty("table") TableHandle table,
             @JsonProperty("outputSymbols") List<Symbol> outputs,
-            @JsonProperty("assignments") Map<Symbol, ColumnHandle> assignments,
-            @JsonProperty("layout") Optional<TableLayoutHandle> tableLayout)
+            @JsonProperty("assignments") Map<Symbol, ColumnHandle> assignments)
     {
         // This constructor is for JSON deserialization only. Do not use.
         super(id);
         this.table = requireNonNull(table, "table is null");
         this.outputSymbols = ImmutableList.copyOf(requireNonNull(outputs, "outputs is null"));
         this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "assignments is null"));
         checkArgument(assignments.keySet().containsAll(outputs), "assignments does not cover all of outputs");
-        this.tableLayout = requireNonNull(tableLayout, "tableLayout is null");
         this.currentConstraint = null;
         this.enforcedConstraint = null;
     }
 
-    public TableScanNode(
+    public static TableScanNode createTableScanNode(
```

Comment:
It's a little weird to have one create as a static function and the other directly using constructor. Two constructors are fine I think.

Reply:
Yeah, but there is two constructor with same parameter.